Tests/test_file_tiff.py

```diff
@@ -450,6 +450,18 @@ def test_save_tiff_with_jpegtables(self):
         # Should not raise UnicodeDecodeError or anything else
         im.save(outfile)
 
+    def test_open_tiff_uint16_multiband(self):
+        """Test opening multiband TIFFs and reading all channels."""
+        base_value = 4660
+        for i in range(2, 6):
+            infile = "Tests/images/uint16_{}_{}.tif".format(i, base_value)
+            im = Image.open(infile)
+            im.load()
+            for j in range(i):
+                current_value = base_value + j
+                im.seek(j)
+                self.assertEqual(im.getpixel((0, 0)), current_value)
```

Comment:
No, there is literally no support for multiband images > 8bpp.  Meaning, there's nothing in the unpacking,  storage or access layers, let alone the various manipulations that would be required to have meaningful support of these images. 

At the very least, it's going to require changes to Unpack.c, Pack.c, Access.c, Storage.c, Convert.c, and likely anywhere the IMAGE_TYPE_INT32 macro shows up.  The reason that it hasn't been done before is more a matter that there's been one with the background, need, and time to get it done.